server/web/api/register.py

```diff
@@ -69,58 +97,57 @@ def login():
 @api.get("/templates")
 @admin_only
 def get_templates():
-    return [
-        {"id": 10023, "name": "Busunfall", "players": 5},
-        {"id": 900323, "name": "Explosion im Wohnviertel", "players": 10},
-    ]
+    return [{"id": scenario.id, "name": scenario.name, "executions": [execution.id for execution in scenario.executions]}
+            for scenario in models.Scenario.query]
 
 
 @api.post("/scenario/start")
-@admin_only
+# @admin_only
+@csrf.exempt
 def start_scenario():
     try:
-        id = request.form["id"]
+        id = int(request.form["id"])
     except KeyError:
         return {"error": "Missing id in request"}, 400
 
-    # TODO: Create actual execution
-    exec_id = math.floor(random() * 1000)
-    test_execution.update({
-        "id": exec_id,
-        "status": "",
-        "players": [
-            {
-                "tan": x,
-                "name": "Max Mustermann",
-                "status": choice(["", "In Vorbereitung"]),
-                "action": "Legt Zugang",
-            }
-            for x in [str(x) for x in uniques(5)]
-        ],
-    })
-    return {
-        "id": exec_id,
-    }
+    if load_execution(id):
+        return active_executions[id].to_dict()
+
+    # Failure
+    raise Exception(f"Could not load execution with id {id}")
 
 
 @api.get("/execution/<int:id>")
 @admin_only
 def get_execution_status(id: int):
-    test_execution["id"] = id
-    random_player = choice(test_execution["players"])
-    random_player["status"] = "In Vorbereitung" if random_player["status"] == "" else ""
-    return test_execution
+    execution = try_get_execution(id)
+    return execution.to_dict()
 
 
 @api.post("/execution/<int:id>/start")
 @admin_only
 def start_execution(id: int):
-    test_execution["status"] = "Aktiv"
-    return test_execution
+    execution = try_get_execution(id)
+    execution.status = Execution.Status.RUNNING
+    return Response(status=204)
 
 
 @api.post("/execution/<int:id>/stop")
 @admin_only
 def stop_execution(id: int):
-    test_execution["status"] = ""
-    return test_execution
+    execution = try_get_execution(id)
+    execution.status = Execution.Status.FINISHED
+    return Response(status=204)
+
+
+@api.post("/execution/<int:id>/player/<tan>/status")
+@admin_only
+def change_player_status(id: int, tan: str):
+    try:
+        status: bool = request.form["alerted"] == "1"
```

Comment:
if u send `"True"` or `"False"` in the form, you can use bool(request.form["alerted"]). Then you dont have to hardcode your expected value.

Reply:
just realized my suiggestion is invalid. consider it resolved.

Reply:
I have implemented an equivalent way by using [python-string-util's booleanize](https://python-string-utils.readthedocs.io/en/develop/manipulation.html?highlight=booleanize#string_utils.manipulation.booleanize)